snappy/click.go

```diff
@@ -496,6 +543,18 @@ func (m *packageYaml) addPackageBinaries(baseDir string) error {
 		if err := helpers.AtomicWriteFile(generateBinaryName(m, binary), []byte(content), 0755, 0); err != nil {
 			return err
 		}
+
+		// now generate a desktop file (if needed)
+		desktop, err := generateSnapDesktopFile(binary, realBaseDir, m)
+		if err != nil {
+			return err
+		}
+		// only write if there is anything for us)
+		if desktop != "" {
+			if err := helpers.AtomicWriteFile(generateDesktopFileName(m, binary), []byte(desktop), 0644, 0); err != nil {
```

Comment:
Should this check to see if the directory exists, and bail if not?  Might solve the prior concern for the time being.